src/main/java/com/jpr/maintenance/database/service/UserService.java

```diff
@@ -0,0 +1,28 @@
+package com.jpr.maintenance.database.service;
+
+import com.jpr.maintenance.database.model.UserEntity;
+import com.jpr.maintenance.database.repository.UserRepository;
+import com.jpr.maintenance.validation.model.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+
+    public Mono<UserEntity> findByUser(User user) {
+        return userRepository.findByUserNameAndPassword(user.username(), user.password());
```

Comment:
This is perhaps a good place to use the User model object, so we can filter out the password when returning the User.